hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueContainer.java

```diff
@@ -958,7 +946,8 @@ public ScanResult scanMetaData() throws InterruptedException {
   public boolean shouldScanData() {
     boolean shouldScan =
```

Comment:
Should we switch shouldScan to be true if it is not open?

Reply:
I think an allow-list like we have here is easier to follow. There's other cases like CLOSING that might be ok, although those should move to CLOSED soon. RECOVERING we would still want to skip. I think DELETED may also be a state encountered on the DN as part of the atomic container delete code but I'd need to refresh my memory on how that was implemented.